octopoes/octopoes/models/path.py

```diff
@@ -100,7 +103,10 @@ def __str__(self) -> str:
         segments = ".".join(map(str, self.segments))
         return f"{start_type}.{segments}"
 
-    def __eq__(self, other: Path):
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, Path | str):
```

Comment:
You probably want a tuple of `Path` and `str`, otherwise you're checking against an instance of a `Union[Path, str]`
```suggestion
        if not isinstance(other, (Path, str)):
```

Reply:
`isinstance` supports Union as argument, see https://docs.python.org/3/library/functions.html#isinstance

Reply:
Ah you're right, this changed in version 3.10 👍